src/_version_check.sh

```diff
@@ -0,0 +1,187 @@
+          
+          # IF LATEST_STABLE >= 2.1
+          else
+            debug "Stable version is greater than $breaking_ver. Running contiguous update"
+            contiguous_update=true
+        fi
+      # If the JWT token is empty for on-prem envs
+      else
+        contiguous_update=false
+        error "Unable to validate versioned images from DockerHub. Likely On-prem or Air-gapped"
+        msg "-------"
+        error "Beginning with version 2.0, PlexTrac is going to require contiguous updates to ensure code migrations are successful and enable us to continue to move forward with improving the platform. We recommend updating to the next minor version available compared to the running version $running_backend_version"
+        error "Are you sure you want to update to $UPGRADE_STRATEGY? (y/n)"
+        get_user_approval hardcheck
+    fi
+
+    upstream_tags=()
+    page=1
+    if [ "$contiguous_update" = true ]
+      then
+        # Get upstream tag list
+        debug "Looking for Running version $running_ver or Breaking version $breaking_ver"
+        while [ $page -lt 600 ]
+          do
+            # Get the available versions from DockerHub and save to array
+            if [[ $(echo "${upstream_tags[@]}" | grep "$breaking_ver" || true) ]]
+              then
+                debug "Found breaking version $breaking_ver"; break;
+            elif [[ $(echo "${upstream_tags[@]}" | grep "$running_ver" || true) ]]
+              then
+                debug "Found running version $running_backend_version"; break;
+            else
+              upstream_tags+=(`wget --header="Authorization: JWT "${JWT_TOKEN} -O - "https://hub.docker.com/v2/repositories/plextrac/plextracapi/tags/?page=$page&page_size=1000" -q | jq -r .results[].name | grep -E '(^[0-9]\.[0-9]*$)' || true`)
+            fi
+            page=$[$page+1]
+        done
+        # Compare stable vs latest_tag and if stable is older than the newest tag, then remove the newest tag to ensure we don't update past stable
+        if [ $stable_date -le $latest_date ]; then
+          debug "Stable is older than latest version"
+          # If stable is older, remove latest from upgrade path
+          for i in ${!upstream_tags[@]}
+            do
+              if [[ ${upstream_tags[i]} = $latest_ver ]]
+                then
+                  debug "correcting upstream_tags to remove latest"
+                  unset 'upstream_tags[i]'
+              fi
+          done
+        else
+          debug "Stable is pinned to the latest version $latest_ver"
+        fi
+        
+        # Remove the running version from the Upgrade path
+        for i in ${!upstream_tags[@]}
+          do
+            if [[ ${upstream_tags[i]} = $running_ver ]]
+              then
+                debug "correcting upstream_tags to remove running version"
+                unset 'upstream_tags[i]'
+            fi
+        done
+        for i in "${!upstream_tags[@]}"; do
```

Comment:
What wizardry is this? Sorting the list?

Reply:
This does in-fact sort the list ⚠️ 🪀